tools/topology/topology2/cavs-nocodec-multicore.conf

```diff
@@ -0,0 +1,473 @@
+		io_clk			$MCLK
+
+		Object.Base.hw_config.1 {
+			name	"SSP2"
+			id	0
+			bclk_freq	3072000
+			tdm_slot_width	32
+			# TODO: remove this. Needs alsaptlg change.
+			Object.Base.link_config.1 {
+				clock_source	1
+			}
+		}
+	}
+]
+
+#
+# Pipeline definitions
+#
+# PCM0 ---> gain ---> Mixin ---> Mixout ---> gain ---> SSP0 (core SSP0_CORE_ID)
+# PCM1 ---> gain ---> Mixin ---> Mixout ---> gain ---> SSP1 (core SSP1_CORE_ID)
+# PCM2 ---> gain ---> Mixin ---> Mixout ---> gain ---> SSP2 (core SSP2_CORE_ID)
+# SSP0 ---> PCM0
+# SSP1 ---> PCM1
+# SSP2 ---> PCM2
+
+# Pipeline ID:1 PCM ID: 0
+Object.Pipeline.host-copier-gain-mixin-playback [
+	{
+		index 1
+
+		core $SSP0_CORE_ID
+		Object.Widget.copier.1 {
+			core_id $SSP0_CORE_ID
+			stream_name 'SSP0 Playback'
+		}
+		Object.Widget.gain.1 {
+			core_id $SSP0_CORE_ID
+			Object.Control.mixer.1 {
+				name 'Playback Volume 1'
+			}
+		}
+	}
+	{
+		index 5
+
+		core $SSP2_CORE_ID
+		Object.Widget.copier.1 {
+			core_id $SSP2_CORE_ID
+			stream_name 'SSP2 Playback'
+		}
+		Object.Widget.gain.1 {
+			core_id $SSP2_CORE_ID
+			Object.Control.mixer.1 {
+				name 'Playback Volume 5'
+			}
+		}
+		Object.Widget.mixin.1 {
+			core_id $SSP2_CORE_ID
+		}
+	}
+]
+
+Object.Pipeline.mixout-gain-dai-copier-playback [
+	{
+		index 14
+
+		core $SSP0_CORE_ID
+		Object.Widget.copier.1 {
+			core_id $SSP0_CORE_ID
+			dai_index	0
+			dai_type "SSP"
+			copier_type "SSP"
+			stream_name "NoCodec-0"
+			node_type $I2S_LINK_OUTPUT_CLASS
+		}
+
+		Object.Widget.gain.1 {
+			core_id $SSP0_CORE_ID
+			Object.Control.mixer.1 {
+				name 'Main Playback Volume 14'
+			}
+		}
+	}
+	{
+		index 6
+
+		core $SSP2_CORE_ID
+		Object.Widget.copier.1 {
+			core_id $SSP2_CORE_ID
+			dai_index	2
+			dai_type "SSP"
+			copier_type "SSP"
+			stream_name "NoCodec-2"
+			node_type $I2S_LINK_OUTPUT_CLASS
+		}
+
+		Object.Widget.gain.1 {
+			core_id $SSP2_CORE_ID
+			Object.Control.mixer.1 {
+				name 'Main Playback Volume 6'
+			}
+		}
+		Object.Widget.mixout.1 {
+			core_id $SSP2_CORE_ID
+		}
+	}
+]
+
+Object.Pipeline.host-gateway-capture [
+	{
+		index 7
+
+		Object.Widget.copier.1 {
+			stream_name 'SSP0 Capture'
+		}
+	}
+	{
+		index 11
+
+		Object.Widget.copier.1 {
+			stream_name 'SSP2 Capture'
+		}
+	}
+]
+
+Object.Pipeline.io-gateway-capture [
+	{
+		index 8
+		direction capture
+
+		Object.Widget.copier."1" {
+			dai_index	1
+			dai_type	"SSP"
+			type		dai_out
+			copier_type	"SSP"
+			stream_name	"NoCodec-0"
+			node_type	$I2S_LINK_INPUT_CLASS
+			Object.Base.audio_format.1 {
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_bit_depth		32
+				out_valid_bit_depth	32
+			}
+		}
+	}
+	{
+		index 12
+		direction capture
+
+		Object.Widget.copier."1" {
+			dai_index	2
+			dai_type	"SSP"
+			type		dai_out
+			copier_type	"SSP"
+			stream_name	"NoCodec-2"
+			node_type	$I2S_LINK_INPUT_CLASS
+			Object.Base.audio_format.1 {
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_bit_depth		32
+				out_valid_bit_depth	32
```

Comment:
@ranj063 do we need to change the formats here?

Reply:
Yes to for clarity but its probably ok for now as it wont break anything